dcpy/utils/data.py

```diff
@@ -81,56 +83,39 @@ def read_data_to_df(
+            )
+            gdf = (
+                df if not data_format.geometry else df_to_gdf(df, data_format.geometry)
+            )
 
-            if not csv.geometry:
-                gdf = df
-
-            else:
-                # case when geometry is in one column (i.e. polygon or point object type)
-                if isinstance(csv.geometry.geom_column, str):
-                    geom_column = csv.geometry.geom_column
-                    assert (
-                        geom_column in df.columns
-                    ), f"❌ Geometry column specified in recipe template does not exist in {local_data_path.name}"
-
-                    # replace NaN values with None. Otherwise gpd throws an error
-                    if df[geom_column].isnull().any():
-                        df[geom_column] = df[geom_column].astype(object)
-                        df[geom_column] = df[geom_column].where(
-                            df[geom_column].notnull(), None
-                        )
-
-                    df[geom_column] = gpd.GeoSeries.from_wkt(df[geom_column])
-
-                    gdf = gpd.GeoDataFrame(
-                        df,
-                        geometry=geom_column,
-                        crs=csv.geometry.crs,
-                    )
-                # case when geometry is specified as lon and lat columns
-                else:
-                    x_column = csv.geometry.geom_column.x
-                    y_column = csv.geometry.geom_column.y
-                    assert (
-                        x_column in df.columns and y_column in df.columns
-                    ), f"❌ Longitude or latitude columns specified in the recipe template do not exist in {local_data_path.name}"
-
-                    gdf = gpd.GeoDataFrame(
-                        df,
-                        geometry=gpd.points_from_xy(df[x_column], df[y_column]),
-                        crs=csv.geometry.crs,
-                    )
         case file.Json():
-            with open(local_data_path) as f:
-                json_str = json.load(f)
-            if data_format.normalize:
-                gdf = pd.json_normalize(json_str, **data_format.normalize)
+            if isinstance(data_format.json_read_meta, file.ReadJson):
+                df = pd.read_json(
+                    local_data_path, **data_format.json_read_meta.json_read_kwargs
+                )
             else:
-                gdf = pd.read_json(json_str)
+                with open(local_data_path) as f:
+                    json_str = json.load(f)
+                df = pd.json_normalize(
+                    json_str, **data_format.json_read_meta.json_read_kwargs
+                )
+
             # Feels a tad hacky to have this here instead of in preprocessing
             # But json columns were causing parquet issues
             if data_format.columns:
-                gdf = gdf[data_format.columns]
+                df = df[data_format.columns]
```

Comment:
If we're coercing any remaining json fields to strings, I'd say we can drop this and let preprocessing take care of it.

Though I'm not seeing where you're doing that?

Reply:
nypl_libraries is the dataset that needs this, so you can use that to test.

Also, an fyi - I changed the cli slightly. So it'll be
- `python3 -m dcpy.lifecycle.ingest.run nypl_libraries -s` on your branch right now
- `python3 -m dcpy.cli lifecycle ingest nypl_libraries -s` if you rebase main into your branch

Reply:
Successfully extracted and converted to parquet:

<img width="1050" alt="image" src="https://github.com/NYCPlanning/data-engineering/assets/144725249/6d65626c-f4c5-46be-ae8e-2aa55d282e60">